api/odigos/v1alpha1/instrumentatioconfig_types.go

```diff
@@ -41,16 +42,76 @@ type SdkConfig struct {
 
 	// configurations for the instrumentation libraries the the SDK should use
 	InstrumentationLibraryConfigs []InstrumentationLibraryConfig `json:"instrumentationLibraryConfigs"`
+
+	// HeadSamplingConfig is a set sampling rules.
+	// This config currently only applies to root spans.
+	// In the Future we might add another level of configuration base on the parent span (ParentBased Sampling)
+	HeadSamplingConfig HeadSamplingConfig `json:"headSamplerConfig,omitempty"`
+}
+
+// 'Operand' represents the attributes and values that an operator acts upon in an expression
+type AttributeCondition struct {
+	// attribute key (e.g. "url.path")
+	Key attribute.Key `json:"key"`
+	// currently only string values are supported.
+	Val string `json:"val"`
+	// The operator to use to compare the attribute value.
+	Operator Operator `json:"operator"`
+}
+
+type Operator string
+
+const (
+	Equals    Operator = "equals"
+	NotEquals Operator = "notEquals"
+	Contain   Operator = "contain"
+	EndWith   Operator = "endWith"
+	StartWith Operator = "startWith"
+)
+
+// AttributesAndSampler is a set of attribute compare samplers that are ANDed together.
+// If all attribute compare samplers evaluate to true, the AND sampler evaluates to true,
+// and the fraction is used to determine the sampling decision.
+// If any of the attribute compare samplers evaluate to false,
+// the fraction is not used and the rule is skipped.
+//
+// An "empty" AttributesAndSampler with no operands is considered to always evaluate to true.
+// and the fraction is used to determine the sampling decision.
+type AttributesAndSamplerRule struct {
+	AttributeConditions []AttributeCondition `json:"operands"`
+	// The fraction of spans to sample, in the range [0, 1].
+	// If the fraction is 0, no spans are sampled.
+	// If the fraction is 1, all spans are sampled.
+	Fraction float64 `json:"fraction"`
+}
+
+// HeadSamplingConfig is a set of attribute rules.
+// The first attribute rule that evaluates to true is used to determine the sampling decision based on its fraction.
+//
+// If none of the rules evaluate to true, the fallback fraction is used to determine the sampling decision.
+type HeadSamplingConfig struct {
+	AttributesAndSamplerRules []AttributesAndSamplerRule `json:"rules"`
+	// Used as a fallback if all rules evaluate to false,
+	// it may be empty - in this case the default value will be 1 - all spans are sampled.
+	// it should be a float value in the range [0, 1] - the fraction of spans to sample.
+	// a value of 0 means no spans are sampled if none of the rules evaluate to true.
+	FallbackFraction *float64 `json:"fallbackFraction,omitempty"`
 }
 
 type InstrumentationLibraryConfig struct {
 	// The name of the instrumentation library
 	// - Node.js: The name of the npm package: `@opentelemetry/instrumentation-<name>`
```

Comment:
This comment should be updated to describe the ID and it's differences between the SDKs